File: Claude/1.Algoritmo Básico Ordenamiento por Burbuja/c++/burbuja.cpp
#include <iostream>
#include <vector>
#include <algorithm>

class BubbleSort {
public:
    /**
     * Implementación del algoritmo de ordenamiento burbuja
     * @param arr Vector de enteros a ordenar
     * @return Vector ordenado
     */
    static std::vector<int> sort(std::vector<int> arr) {
        int n = arr.size();
        
        for (int i = 0; i < n - 1; i++) {
            bool swapped = false;
            for (int j = 0; j < n - i - 1; j++) {
                if (arr[j] > arr[j + 1]) {
                    std::swap(arr[j], arr[j + 1]);
                    swapped = true;
                }
            }
            if (!swapped) break;
        }
        return arr;
    }
};

int main() {
    std::vector<int> numbers = {64, 34, 25, 12, 22, 11, 90};
    std::vector<int> original = numbers;
    std::vector<int> sorted = BubbleSort::sort(numbers);
    
    std::cout << "Array original: ";
    for (int num : original) std::cout << num << " ";
    std::cout << "\nArray ordenado: ";
    for (int num : sorted) std::cout << num << " ";
    std::cout << std::endl;
    
    return 0;
}

File: Gemini/1. Algoritmo Básico Ordenamiento por Burbuja/Typescript/burbuja.ts
//npm i --save-dev @types/node
import * as readline from 'readline';

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

function bubbleSort(arr: number[]): number[] {
  const n = arr.length;
  for (let i = 0; i < n - 1; i++) {
    for (let j = 0; j < n - 1 - i; j++) {
      // Compara elementos adyacentes
      if (arr[j] > arr[j + 1]) {
        // Intercambia los elementos si están en el orden incorrecto
        const temp = arr[j];
        arr[j] = arr[j + 1];
        arr[j + 1] = temp;
      }
    }
  }
  return arr;
}

console.log("--- Algoritmo de Ordenamiento por Burbuja ---");
rl.question('Ingresa números enteros separados por comas (ej: 5,2,8,1): ', (input) => {
  try {
    // Convierte el string de entrada en un arreglo de números
    const numeros = input.split(',').map(item => {
      const num = parseInt(item.trim(), 10);
      if (isNaN(num)) {
        throw new Error(`'${item.trim()}' no es un número válido.`);
      }
      return num;
    });

    console.log("Arreglo original:", numeros);
    const arregloOrdenado = bubbleSort([...numeros]); // Se pasa una copia para no modificar el original
    console.log("Arreglo ordenado:", arregloOrdenado);

  } catch (error) {
    if (error instanceof Error) {
        console.error("Error:", error.message);
    } else {
        console.error("Ocurrió un error inesperado.");
    }
  } finally {
    rl.close();
  }
});